altair/vegalite/v5/api.py

```diff
@@ -380,26 +425,727 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
+    else:
+        if not isinstance(statement, t.Mapping):
+            statement = _parse_literal(statement, str_as)
+        selection = conditions
+        selection.update(**statement, **kwds)  # type: ignore[call-arg]
+    return selection
+
+
+class _BaseWhen(Protocol):
+    # NOTE: Temporary solution to non-SchemaBase copy
+    _condition: _ConditionType
+
+    def _when_then(
+        self,
+        statement: _StatementOrLiteralType,
+        kwds: dict[str, Any],
+        str_as: _StrAsType,
+    ) -> _ConditionClosed | _Condition:
+        condition: Any = _deepcopy(self._condition)
+        then = _parse_then(statement, kwds, str_as)
+        condition.update(then)
+        return condition
+
+
+class When(_BaseWhen):
+    """Utility class for ``when-then-otherwise`` conditions.
+
+    Represents the state after calling :func:`.when()`.
+
+    This partial state requires calling :meth:`When.then()` to finish the condition.
+
+    References
+    ----------
+    `polars.expr.whenthen <https://github.com/pola-rs/polars/blob/b85c5e0502ca99c77742ee25ba177e6cd11cf100/py-polars/polars/expr/whenthen.py>`__
+    """
+
+    def __init__(self, condition: _ConditionType, /) -> None:
+        self._condition = condition
+
+    @overload
+    def then(
+        self, statement: str, *, str_as: Literal["value"] = ..., **kwds: Any
+    ) -> Then[_Conditions]: ...
+    @overload
+    def then(
+        self, statement: str, *, str_as: Literal["shorthand"], **kwds: Any
+    ) -> Then[_Condition]: ...
+    @overload
+    def then(
+        self,
+        statement: _Value | _LiteralNumeric | Sequence[_LiteralNumeric],
+        **kwds: Any,
+    ) -> Then[_Conditions]: ...
+    @overload
+    def then(
+        self, statement: dict[str, Any] | SchemaBase, **kwds: Any
+    ) -> Then[Any]: ...
+    def then(
+        self,
+        statement: _StatementOrLiteralType,
+        *,
+        str_as: _StrAsType = "value",
+        **kwds: Any,
+    ) -> Then[Any]:
+        """Attach a statement to this predicate.
+
+        Parameters
+        ----------
+        statement
+            A spec or value to use when the preceding :func:`.when()` clause is true.
+        str_as
+            Wrap strings in :func:`.value()` or encode as `shorthand<https://altair-viz.github.io/user_guide/encodings/index.html#encoding-shorthands>`__.
+
+            .. note::
+                ``str_as="shorthand"`` is the behavior used in :func:`.condition()`.
+        **kwds
+            Additional keyword args are added to the resulting ``dict``.
+
+        Returns
+        -------
+        :class:`Then`
+        """
+        condition = self._when_then(statement, kwds, str_as)
+        if _is_condition_extra(condition, statement, kwds=kwds, str_as=str_as):
+            return Then(_Conditional(condition=condition))
+        else:
+            return Then(_Conditional(condition=[condition]))
+
+
+class Then(core.SchemaBase, t.Generic[_C]):
```

Comment:
Following https://github.com/vega/altair/commit/d12a74117ec201fc096f5e27328d4e50ea493f84 `Then` inherits from `SchemaBase`.

This addresses https://github.com/vega/altair/pull/3427#issuecomment-2209536374

@mattijn 
> If I don't specify a `.otherwise()`, I get the following output
> ```py
> >>> alt.when(brush).then('count()')#.otherwise('grey')
> ```
> <altair.vegalite.v5.api.Then at 0x2c24399d960>
> 
> 
> But at the above mentioned VL page on condition it is mentioned that:

> > Note that if the “else” case value is not specified, default mark color will be applied.

> I noticed it work correctly within an `alt.Chart()` defintion, as in, it doesnt error when `.otherwise()` is not included.

A drawback to this is the `Then.[when|otherwise]` methods are more difficult to find with auto-complete:

## When

![image](https://github.com/vega/altair/assets/125183946/68cc96f4-f3de-4c19-91b1-bc4d96858685)


## Then

![image](https://github.com/vega/altair/assets/125183946/55622c86-7a7c-401c-bb3a-a440be8b5d48)


## ChainedWhen

![image](https://github.com/vega/altair/assets/125183946/7202e1bc-0577-4ab7-a44d-a4f0774f4529)